File: Two Sum II - Input array is sorted/Two Sum II - Input array is sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];
        int left=0, right=numbers.length-1;
        while(left<right){
            if(numbers[left]+numbers[right]>target){
                while(left<right && numbers[right]==numbers[right-1]) right--;
                right--;
            }
            else if(numbers[left]+numbers[right]<target){
                while(left<right && numbers[left]==numbers[left+1]) left++;
                left++;
            }
            else{
                ans[0]=left+1;
                ans[1]=right+1;
                break;
            }
        }
        return ans;
    }
}

File: Two Sum II - Input array is sorted/Two Sum II - Input array is sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans = []
        l, r = 0, len(numbers)-1
        while l<r:
            if numbers[l]+numbers[r]>target:
                while l<r and numbers[r]==numbers[r-1]:
                    r-=1
                r-=1
            elif numbers[l]+numbers[r]<target:
                while l<r and numbers[l]==numbers[l+1]:
                    l+=1
                l+=1
            else:
                ans+=[l+1, r+1]
                break
        return ans